src/components/VmNics/index.js

```diff
@@ -39,15 +38,15 @@ class VmNic extends React.Component {
     const idPrefix = `vmnic-${nic.get('name')}`
     const nicInfoContent = (
       <div id={`${idPrefix}-nic-info`}>
-        {msg.network()}: {vnicProfile.getIn(['network', 'name'])}
+        {msg.network()}: {vnicProfile ? vnicProfile.getIn(['network', 'name']) : msg.notConnected()}
         <br />
-        {msg.vnicProfile()}: {vnicProfile.get('name')}
+        {msg.vnicProfile()}: {vnicProfile ? vnicProfile.get('name') : msg.notConnected()}
       </div>
     )
 
     const text = (
       <span className={style['light']} id={`${idPrefix}-nic`}>
-        ({vnicProfile ? `${vnicProfile.getIn(['network', 'name'])}/${vnicProfile.get('name')}` : null})
+        ({vnicProfile ? `${vnicProfile.getIn(['network', 'name'])}/${vnicProfile.get('name')}` : `${msg.notConnected()}/${msg.notConnected()}`})
```

Comment:
Wouldn't it be better to render more descriptive test instead of double `Not connected/Not connected`?

Reply:
btw it also needs an option to set the network to "no network" when creating/editing a NIC.